spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -103,20 +103,23 @@ object DistriOptimizer {
     trainSummary: Option[TrainSummary],
     validationSummary: Option[ValidationSummary],
     isOverWrite: Boolean
-  )(implicit ev: TensorNumeric[T]) = {
+  )(implicit ev: TensorNumeric[T]): Unit = {
     val sc = dataset.originRDD().sparkContext
     val partitionNum = dataset.originRDD().partitions.length
     var wallClockTime = 0L
     var lastEpochTime = 0L
-    val driverState = T("epoch" -> optimMethod.state.get[Int]("epoch").getOrElse(1),
+    val driverState = T(
+      "epoch" -> optimMethod.state.get[Int]("epoch").getOrElse(1),
       "neval" -> optimMethod.state.get[Int]("neval").getOrElse(1),
       "Loss" -> optimMethod.state.get[Float]("Loss").getOrElse(Float.PositiveInfinity),
-      "score" -> optimMethod.state.get[Float]("score").getOrElse(0f))
-    val _subModelNumber = Engine.getEngineType match {
+      "score" -> optimMethod.state.get[Float]("score").getOrElse(0f)
+    )
+    val _subModelNumber = Engine.getEngineType() match {
       case MklBlas => coresPerNode
-      case _ => throw new IllegalArgumentException()
     }
-    var accumulateCount = 0
+
+    val numSamples = dataset.data(train = false).map(_.size()).reduce(_ + _)
```

Comment:
I just test the time cost for this counting on Imagenet DataSet on 8 E5-haswell nodes.
```
2017-08-03 09:41:03 INFO  DistriOptimizer$:121 - Counting samples...
2017-08-03 09:41:36 INFO  DistriOptimizer$:125 - Total: 1281167. Takes 32.803411577s
```
32s seems not long compared with the looading time and training time.
Could you add a log info for this counting?